tests/smoke.rs

```diff
@@ -110,8 +111,65 @@ fn rng_init() {
 #[test]
 #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
 fn with_seed() {
-    let a = fastrand::Rng::with_seed(7);
-    let b = fastrand::Rng::new();
+    let a = fastrand::Rng::<CellSeed>::with_seed(7);
+    let b = fastrand::Rng::<CellSeed>::new();
     b.seed(7);
     assert_eq!(a.u64(..), b.u64(..));
 }
+
+
+
+mod atomic {
+    use std::sync::atomic::{AtomicU64, Ordering};
+    use fastrand::{Rng, Seed};
+
+    struct AtomicSeed(AtomicU64);
+    impl Seed for AtomicSeed {
+        fn with_seed(seed: u64) -> Rng<Self> where Self: Sized {
+            Rng::from(AtomicSeed(AtomicU64::new(seed)))
+        }
+
+        fn get(&self) -> u64 {
+            self.0.load(Ordering::Relaxed)
+        }
+
+        fn set(&self, new_seed: u64) {
+            self.0.store(new_seed, Ordering::Relaxed)
+        }
```

Comment:
If thread B reads a value between the time thread A reads it and the time it stores the value, thread A and thread B will get the same value from rng. I guess there are both use cases where that is ok and use cases where it is not ok, but I'm not positive about having an API that is easy to misuse.

(If you want to create atomic rng, I think you need to replace get/set in gen_u64 with fetch_add)